api-report/container-runtime.api.md

```diff
@@ -657,6 +668,9 @@ export class PendingStateManager implements IDisposable {
     replayPendingStates(): void;
 }
 
+// @public
+export function requestSummarizer(loader: ILoader, lastSequenceNumber: number, { cache, reconnect, summarizingClient, url }: ISummarizerRequestOptions): Promise<ISummarizer>;
```

Comment:
Can you please check why we need lastSequenceNumber?
Summarizer always connects as "write", meaning that it will always process all the ops that exists in the file (own join op works as a barrier here). So there is really no value in passing sequence number through pipeline given that no matter what it's value, we will summarize past that.

Or maybe I misunderstand how it works? Maybe somewhere we pause op processing? I do not think we have that capability, as we can't load random snapshot from the past...

Reply:
It's used in Loader.resolveCore() to let/make the container "catch up" to the specified sequence number before returning it to the caller:

        if (container.deltaManager.lastSequenceNumber <= fromSequenceNumber) {
            await new Promise<void>((resolve, reject) => {
                function opHandler(message: ISequencedDocumentMessage) {
                    if (message.sequenceNumber > fromSequenceNumber) {
                        resolve();
                        container.removeListener("op", opHandler);
                    }
                }

                container.on("op", opHandler);
            });
        }

So not passing it would mean just accepting the container's current state, whatever that happens to be. I don't see a problem with that, but I wonder why it was written this way originally.

Reply:
I believe it should not matter, as summarizer always waits to be connected, which means it will always catch up to latest state. So I'd rather remove all of that logic and seq# passing - it just does not make any sense to have it there - makes API cleaner.

Reply:
That works for me.